venafi/resource_venafi_certificate_test.go

```diff
@@ -1018,52 +1150,299 @@ func TestCloudCsrService(t *testing.T) {
+	t.Logf("Testing importing TPP cert:\n %s", config)
+	r.Test(t, r.TestCase{
+		Providers: testProviders,
+		Steps: []r.TestStep{
+			r.TestStep{
+				Config:        config,
+				ResourceName:  "venafi_certificate.token_tpp_certificate_import",
+				ImportStateId: importId,
+				ImportState:   true,
+				ImportStateCheck: func(states []*terraform.InstanceState) error {
+					t.Log("Importing TPP certificate with CSR Service Generated", data.cn)
+					return checkStandardImportCert(t, data, states)
+				},
+			},
+		},
+	})
+}
 
-	t.Logf("Testing private key PEM:\n %s", privateKey)
-	privateKeyString, err := util.DecryptPkcs8PrivateKey(privateKey, data.private_key_password)
+func TestImportCertificateCloudPartCreate(t *testing.T) {
+	data := getCertCloudImportConfig()
+	config := fmt.Sprintf(cloudCsrServiceConfigImportCreate, cloudProviderImport, data.cn, data.key_algo, data.private_key_password, data.expiration_window)
+	t.Logf("Creating VaaS Token certificate with Service CSR generated and config to be be imported on next test:\n %s", config)
+	r.Test(t, r.TestCase{
+		Providers: testProviders,
+		Steps: []r.TestStep{
+			r.TestStep{
+				Config: config,
+				Check: func(s *terraform.State) error {
+					t.Log("Issuing VaaS certificate with CSR Service Generated", data.cn)
+					err := checkStandardCertPKCS8(t, data, s)
+					if err != nil {
+						return err
+					}
+					err = checkStandardCertId(t, data, s)
+					if err != nil {
+						return err
+					}
+					return nil
+				},
+			},
+		},
+	})
+}
 
-	_, err = tls.X509KeyPair([]byte(certificate), []byte(privateKeyString))
+func TestImportCertificateCloudPartImport(t *testing.T) {
+	// Important!: TestImportCertificateCloudPartCreate must be run before this test to ensure proper import
+	data := getCertCloudImportConfig()
+	vaasIdp, err := getVaasId()
 	if err != nil {
-		return fmt.Errorf("error comparing certificate and key: %s", err)
+		fmt.Errorf(err.Error())
 	}
+	var vaasId string
+	vaasId = *vaasIdp
+	config := fmt.Sprintf(cloudCsrServiceConfigImport, cloudProviderImport)
+	importId := fmt.Sprintf("%s,%s", vaasId, data.private_key_password)
+	t.Logf("Testing importing VaaS cert:\n %s", config)
+	r.Test(t, r.TestCase{
+		Providers: testProviders,
+		Steps: []r.TestStep{
+			r.TestStep{
+				Config:        config,
+				ResourceName:  "venafi_certificate.token_cloud_certificate_import",
+				ImportStateId: importId,
+				ImportState:   true,
+				ImportStateCheck: func(states []*terraform.InstanceState) error {
+					t.Log("Importing VaaS certificate with CSR Service Generated", data.cn)
+					return checkStandardImportCert(t, data, states)
+				},
+			},
+		},
+	})
+}
 
-	return nil
+func getVaasId() (*string, error) {
+	vaasIdBytes, err := getBytesFromFile(vaas_id_path)
+	if err != nil {
+		return nil, err
+	}
+	vaasId := string(vaasIdBytes)
+	return &vaasId, nil
+}
+
+func getBytesFromFile(p string) ([]byte, error) {
+	path := GetAbsoluteFIlePath(p)
+	file, err := os.Open(path)
```

Comment:
you may need to close the file after reading his content, i.e.: defer file.Close()